source/base/parameter_handler.cc

```diff
@@ -1862,6 +1862,7 @@ namespace
 
 bool ParameterHandler::read_input_from_xml (std::istream &in)
 {
+  AssertThrow(in, ExcIO());
```

Comment:
The fact that these `read` functions return a boolean is really a 20-year old artifact from a time when I didn't yet know how to program C++. We should abandon the thing with the return value and instead throw exceptions.